api/auth.js

```diff
@@ -114,28 +114,36 @@ module.exports = {
   },
 
   check: function(req, permissions, districtId, callback) {
-    module.exports.getUserCtx(req, function(err, userCtx) {
-      if (err) {
-        return callback(err);
-      }
-      if (isDbAdmin(userCtx)) {
-        return callback(null, { user: userCtx.name });
-      }
-      if (!module.exports.hasAllPermissions(userCtx, permissions)) {
-        return callback({ code: 403, message: 'Insufficient privileges' });
-      }
-      module.exports.getFacilityId(req, userCtx, function(err, facilityId) {
+    const doIt = (resolve, reject) => {
```

Comment:
Not an easy thing to name. Maybe `doCheck` or `promiseCheck`? IDK.

Reply:
called it `_check`, because why not. Honestly I could just call it `check` again, but I like to avoid shadowing where possible in JS 💀